ethexe/processor/src/lib.rs

```diff
@@ -218,8 +218,9 @@ impl Processor {
     ) -> Result<Vec<LocalOutcome>> {
         log::debug!("Processing upload code {code_id:?}");
 
-        let valid = !(code_id != CodeId::generate(code) || self.handle_new_code(code)?.is_none());
+        let valid = code_id == CodeId::generate(code) && self.handle_new_code(code)?.is_some();
```

Comment:
`&&` is executed lazily, i.e. if the left side is true, then `handle_new_code` will not be called

Reply:
maybe it's worth using Option instead of logical operators here? both logical operators are lazy: https://doc.rust-lang.org/reference/expressions/operator-expr.html#arithmetic-and-logical-binary-operators